devex-1180 - pulling in rosies work on split tests

codyseibert opened the pull request.

codyseibert committed:
devex-1180 - pulling in rosies work on split tests

codyseibert committed:
devex-1180 - trying to split unit tests and coverage

codyseibert committed:
1180 - trying to prevent tests from failing

codyseibert committed:
fixing artifcats

codyseibert committed:
devex-1180 - increase coverage

codyseibert committed:
devex-1180 - reducing number of runners

codyseibert committed:
devex-1180 - final clean up

codyseibert committed:
devex-1180 reduce runners move

rosiehsmith commented on .github/workflows/client.yml:
```diff
@@ -19,4 +26,30 @@ jobs:
+          name: coverage-artifacts
+          path: coverage
+      - name: Process Coverage
+        run: npx nyc report --check-coverage --branches 94.56 --functions 98.81 --lines 99.25 --statements 99.23 --reporter lcov --reporter text --reporter clover -t coverage
```
sweet!

rosiehsmith commented on split-tests-glob.ts:
```diff
@@ -0,0 +1,16 @@
+
+let testFiles;
+if (testType.includes('unit')) {
+  // could use for shared? pass the path?
```
could cleanup this comment i had added

codyseibert committed:
removing a comment

rosiehsmith commented on web-client/jest-unit.config.ts:
```diff
@@ -1,4 +1,7 @@
 import type { Config } from 'jest';
+const testFilesEnv = process.env.TESTFILES;
+const testFiles = testFilesEnv?.split(' ') || [];
+console.log('testFiles in config', testFiles);
```
woops one more log

rosiehsmith committed:
devex-split-shared: splitting shared tests into 4 parallel on CI

rosiehsmith committed:
devex-split-shared: add coverage merging

devex-split-shared: rename artifact path

devex-shared: increase nodes so I can see results faster

devex-shared: update jest config, temporarily run more nodes to see results faster

devex-shared: renamed coverage artifacts

devex-shared: add tsx and jsx to moduleFileExtensions

devex-1180 removing unused lines

Revert "devex-split-shared: rename artifact path"

This reverts commit 1c8b0aecfba8c93e26c0f277fb8117f0543cbba0.

Revert "devex-shared: increase nodes so I can see results faster"

This reverts commit 63795d5d52a2740e2d720a5c9b7dd4accb474b7a.

codyseibert committed:
devex-1180 removing unused lines

codyseibert committed:
Merge pull request #10048 from flexion/devex-split-shared-tests

devex-split-shared: splitting shared tests into 4 parallel on CI

mmarcotte reviewed (approved):
👏 👏 👏 🙏
